src/cljs/main/broadfcui/common/table/table.cljs

```diff
@@ -0,0 +1,140 @@
+
+
+(react/defc Table
+  {:update-query-params
+   (fn [{:keys [state]} new-params]
+     (let [old-state (:query-params @state)
+           new-state (merge old-state new-params)
+           different? (not= old-state new-state)]
+       (when different?
+         (swap! state assoc :query-params new-state))
+       different?))
+   :refresh-rows
+   (fn [{:keys [props state]}]
+     (swap! state assoc :loading? true)
+     ((-> props :table :data-source) {:columns (-> props :table :columns)
+                                      :query-params (:query-params @state)
+                                      :on-done (fn [{:keys [total-count filtered-count results]}]
+                                                 (swap! state assoc
+                                                        :total-count total-count
+                                                        :filtered-count filtered-count
+                                                        :rows results
+                                                        :loading? false))}))
+   :get-initial-state
+   (fn [{:keys [props]}]
+     (assoc
+      (persistence/try-restore
+       {:key (:state-key props)
+        :validator (fn [stored-value]
+                     (or (not (:v props))
+                         (= (:v props) (:v stored-value))))
+        :initial
+        #(let [columns (-> props :table :columns)
+               initial-sort-column (or (first (filter :sort-initial columns))
+                                       (when-not (-> props :table :behavior :allow-no-sort?)
+                                         (first columns)))
+               initial-sort-order (when initial-sort-column
+                                    (get initial-sort-column :sort-initial :asc))]
+           (merge
+            {:query-params (select-keys
+                            {:page-number 1
+                             :rows-per-page 20
+                             :filter-text ""
+                             :sort-column (table-utils/resolve-id initial-sort-column)
+                             :sort-order initial-sort-order}
+                            (difference all-query-params (-> props :table :external-query-params)))
+             :column-display (table-utils/build-column-display columns)}
+            (when-let [v (:v props)] {:v v})))})
+       :rows []))
+   :render
+   (fn [{:keys [props state]}]
+     (let [props (utils/deep-merge default-props props)
+           {:keys [rows column-display total-count filtered-count query-params]} @state
+           {:keys [table toolbar paginator]} props
+           {:keys [empty-message columns behavior external-query-params]} table
+           {:keys [fixed-column-count allow-no-sort?]} behavior
+           query-params (merge query-params (select-keys props external-query-params))
+           update-column-display #(swap! state assoc :column-display %)]
```

Comment:
In lieu of documentation, this might be a good place to make some assertions about the structure of the `:columns` key and fields within. Same goes for any other keys that are crucial.